### ISSUE TITLE
BUILD_INFO export is unused (safe to keep or remove)

### ISSUE THREAD

**Issue by rjwalters** [rulehunt/rulehunt#226]
## Summary

The `BUILD_INFO` constant exported from `src/buildInfo.ts` is flagged as unused by static analysis, though it may be intentionally exported for build metadata purposes.

### Current State
```typescript
// src/buildInfo.ts:4
export const BUILD_INFO = {
  // Build metadata generated at build time
}
```

### Usage Check
```bash
ts-prune --error | grep BUILD_INFO
# Result: src/buildInfo.ts:4 - BUILD_INFO
```

No imports found in:
- Frontend code (src/)
- Backend functions (functions/)
- Tests

### File Generation
The file appears to be generated by:
```json
// package.json:9
"build": "node scripts/generate-build-info.cjs && tsc -b && vite build && pnpm run copy:resources"
```

Script: `scripts/generate-build-info.cjs`

### 1. Build Metadata (Intentionally Unused)
The export might be for:
- Exposing build info in browser devtools
- Future debugging features
- Build verification

### 2. Actually Unused
If not needed, the entire buildInfo system could be removed.

## Investigation Needed

**Before deciding to remove**, check:

1. **Is BUILD_INFO accessed at runtime?**
   ```typescript
   // Check if it's used via window/global object
   // Or imported dynamically
   ```

2. **Is it used in devtools?**
   - Open browser console on deployed site
   - Check for `window.BUILD_INFO` or similar

3. **Is it injected into HTML?**
   - Check Vite build output
   - Look in dist/index.html for build metadata

### Option A: Keep (If Intentional)
If BUILD_INFO is for build metadata/debugging:
- Document its purpose in comments
- Add to .eslintrc as intentionally unused export
- Keep as-is

### Option B: Use It
If it should be exposed for debugging:
```typescript
// src/buildInfo.ts
export const BUILD_INFO = { /* ... */ }

// Expose in dev mode
if (import.meta.env.DEV) {
  window.BUILD_INFO = BUILD_INFO
}
```

### Option C: Remove
If truly unused:
- Remove `src/buildInfo.ts`
- Remove `scripts/generate-build-info.cjs`  
- Update build script in package.json

## Impact

- **If removed**: ~20-30 lines, simpler build process
- **If kept**: Zero impact, just documenting purpose
- **Risk**: Very low either way

## Recommendation

**Investigate first** - Check if this is accessed at runtime before removing. If not used, it's safe to remove the entire buildInfo generation system.

---

**Category**: Potentially unused code (requires investigation)  
**Effort**: 15 minutes investigation + 10 minutes removal (if applicable)  
**Impact**: Very low  
**Priority**: Very low
